### DISCUSSION TITLE
compile error with 3.10.5

### DISCUSSION THREAD

**Discussion by jamesbloom-scheff** [nlohmann/json#3559] [Q&A] [answered]
I have 3.10.5 (just downloaded it), Visual studio 2019. Trying to build for 32 bits. 

My code looks like this

#undef max
#define NOMINMAX
#include "Json.hpp"

I get this

1>C:\Users\bloomscj\source\repos\RESTProject2\nlohmann\detail\meta\type_traits.hpp(641,101): error C2220: the following warning is treated as an error
1>C:\Users\bloomscj\source\repos\RESTProject2\nlohmann\detail\meta\type_traits.hpp(641,101): warning C4003: not enough arguments for function-like macro invocation 'min'
1>C:\Users\bloomscj\source\repos\RESTProject2\nlohmann\detail\meta\type_traits.hpp(641,101): error C2589: '(': illegal token on right side of '::'
1>C:\Users\bloomscj\source\repos\RESTProject2\nlohmann\detail\meta\type_traits.hpp(644): message : see reference to class template instantiation 'nlohmann::detail::value_in_range_of_impl2<OfType,T,true,true>' being compiled
1>C:\Users\bloomscj\source\repos\RESTProject2\nlohmann\detail\meta\type_traits.hpp(641,101): error C2760: syntax error: unexpected token '(', expected ')'

it even worse if I remove the first two lines. I've tried C++ 14, 17, and 20. Didn't make a difference. Happens for both debug and release builds

I've noticed that others had similar problems, but they are all reported as fixed in 3.10.5

Not sure what else to try,

**Comment by jamesbloom-scheff** [comment, marked as answer]
nvm

I had an automatic inclusion of the header file. Evidently, including <curl/curl.h> causes the problem.

making sure that 
#undef max
#define NOMINMAX

are before #include <curl/curl.h> makes the problem go away

**Reply by gregmarr** [reply]
Thanks for the update @jamesbloom-scheff , please mark this as the answer.
